source/heating_model/constant_tidal_heating.cc

```diff
@@ -0,0 +1,126 @@
+/*
+  Copyright (C) 2011 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/heating_model/constant_tidal_heating.h>
+
+
+namespace aspect
+{
+  namespace HeatingModel
+  {
+    template <int dim>
+    void
+    ConstantTidalHeating<dim>::
+    evaluate (const MaterialModel::MaterialModelInputs<dim> &/*material_model_inputs*/,
+              const MaterialModel::MaterialModelOutputs<dim> &/*material_model_outputs*/,
+              HeatingModel::HeatingModelOutputs &heating_model_outputs) const
+    {
+      for (unsigned int q=0; q<heating_model_outputs.heating_source_terms.size(); ++q)
+        {
+          // return a constant value
+          heating_model_outputs.heating_source_terms[q] = 2 * constant_viscosity * pow(constant_tidal_strain_rate, 2.0)
+                                                          / ( 1 + pow( (constant_tidal_frequency * constant_viscosity) / constant_shear_modulus, 2.0 ) );
```

Comment:
```suggestion
          heating_model_outputs.heating_source_terms[q] = 2 * constant_viscosity * std::pow(constant_tidal_strain_rate, 2.0)
                                                          / ( 1 + std::pow( (constant_tidal_frequency * constant_viscosity) / constant_shear_modulus, 2.0 ) );
```

Reply:
In the past, we have found that it is substantially faster if you replace `std::pow(x,2.0)` by `x*x`. It might be useful to do that here too.